bundles/org.openhab.binding.echonetlite/src/main/java/org/openhab/binding/echonetlite/internal/EchonetLiteBridgeHandler.java

```diff
@@ -0,0 +1,399 @@
+                }
+            }
+        }
+    }
+
+    private void pollRequests(long nowMs) {
+        @Nullable
+        Message message;
+        while (null != (message = requests.poll())) {
+            logger.info("Received request: {}", message);
+            if (message instanceof NewDeviceMessage) {
+                newDeviceInternal((NewDeviceMessage) message);
+            } else if (message instanceof RefreshMessage) {
+                refreshDeviceInternal((RefreshMessage) message, nowMs);
+            } else if (message instanceof RemoveDevice) {
+                removeDeviceInternal((RemoveDevice) message);
+            } else if (message instanceof UpdateDevice) {
+                updateDeviceInternal((UpdateDevice) message);
+            } else if (message instanceof StartDiscoveryMessage) {
+                startDiscoveryInternal((StartDiscoveryMessage) message);
+            } else if (message instanceof StopDiscoveryMessage) {
+                stopDiscoveryInternal((StopDiscoveryMessage) message);
+            }
+        }
+    }
+
+    private void pollNetwork(long nowMs, EchonetChannel echonetChannel) {
+        try {
+            echonetChannel.pollMessages(echonetMessage, this::onMessage,
+                    EchonetLiteBindingConstants.NETWORK_WAIT_TIMEOUT);
+        } catch (IOException e) {
+            logger.error("Failed to poll for messages", e);
+        }
+    }
+
+    private void onMessage(final EchonetMessage echonetMessage) {
+        final EchonetClass echonetClass = echonetMessage.sourceClass();
+        final SocketAddress sourceAddress = echonetMessage.sourceAddress();
+        if (null == echonetClass) {
+            logger.warn("Unable to find echonetClass for message: {}", echonetMessage.toDebug());
+            return;
+        }
+
+        if (null == sourceAddress) {
+            logger.warn("Unable to find sourceAddress for message: {}", echonetMessage.toDebug());
+            return;
+        }
+
+        final InstanceKey instanceKey = new InstanceKey((InetSocketAddress) sourceAddress, echonetClass,
+                echonetMessage.instance());
+        final Esv esv = echonetMessage.esv();
+
+        EchonetObject echonetObject = devicesByKey.get(instanceKey);
+        if (null == echonetObject) {
+            echonetObject = devicesByKey.get(discoveryKey);
+        }
+
+        logger.debug("Message {} for: {}", esv, echonetObject);
+        if (null != echonetObject) {
+            echonetObject.applyHeader(esv, echonetMessage.tid());
+            while (echonetMessage.moveNext()) {
+                final int epc = echonetMessage.currentEpc();
+                final int pdc = echonetMessage.currentPdc();
+                ByteBuffer edt = echonetMessage.currentEdt();
+                echonetObject.applyProperty(instanceKey, esv, epc, pdc, edt);
+            }
+        }
+    }
+
+    private void poll() {
+
+        updateStatus(ThingStatus.ONLINE);
```

Comment:
This seems too early since the polling could fail, in which case it would immediately after switch to **OFFLINE**.

Reply:
I've moved this until after one successful poll cycle has been completed.